src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -60,6 +60,14 @@ default String getAsTextShowAll() {
         }
         return builder.toString();
     }
+    
+    default String getPrintableString(Printable... printables){
+    	final StringBuilder builder = new StringBuilder();
+    	for(Printable p : printables) {
+    		builder.append(p.getPrintableString());
```

Comment:
since your individual `getPrintableString()` methods don't format the details in the required way, this method should have handled the formatting as well.